file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/caching/CachedContextProvider.java

```diff
@@ -0,0 +1,40 @@
+package com.file.sharing.core.caching;
+
+import com.file.sharing.core.objects.StorageInfo;
+import com.file.sharing.core.objects.UserInfo;
+import com.file.sharing.core.service.StorageService;
+import com.file.sharing.core.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import static com.file.sharing.core.objects.Constants.STORAGE_INFO_CACHE_NAME;
+import static com.file.sharing.core.objects.Constants.USER_INFO_CACHE_NAME;
+
+/**
+ * created by: andrei
+ * date: 06.04.2019
+ **/
+@Component
+public class CachedContextProvider {
```

Comment:
This name is ambigous. It says it's a context provider, but handels user info and storage info. Just by looking at the attributes and methods of this class, it doesn';t reflect any context capabilities. 

To keep the responsibilities clear and separated, I would create 2 cache services : 
- UserInfoCache
- StorageInfoCache

These cache services will wrap each their respective service (UserService, StorageService) and will have the following methods:
- get
- evict

We will use the evict method later on, together with spring application event capabilies, to evict each cache when an item action is performed : download, upload.... etc...